### PR TITLE
[Admin][Products] Add product properties create/edit flow to admin

### PR DESCRIPTION
**Pull request by MadelineCollier**
## Summary
This change addresses https://github.com/solidusio/solidus/issues/5857

This builds off of the excellent work done by @Astr0surf3r in https://github.com/solidusio/solidus/pull/5885 and by @forkata in https://github.com/solidusio/solidus/pull/5926 to build out the create/edit/update flow for product properties in the new admin.

I've added a request spec that was missing and tweaked a few other things.

## Screenshots

https://github.com/user-attachments/assets/a1ce960d-a051-4c74-a998-666bdff71800

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

**Comment by tvdeyen** [comment]
@MadelineCollier can we wait until #6014 has been finalized and merged? This contains everything we need to use semantic links and form redirects. I am happy to help update this PR (it's not much actually) afterwards.

**Comment by MadelineCollier** [comment]
> @MadelineCollier can we wait until #6014 has been finalized and merged? This contains everything we need to use semantic links and form redirects. I am happy to help update this PR (it's not much actually) afterwards.

For sure, I can hold off and integrate it later today (assuming the other PR is merged today). Thanks for your work on solving that issue by the way.

**Review by kennyadsl — APPROVED** [review]

**Pull request merged**
